lxd/containers_get.go

```diff
@@ -43,9 +44,67 @@ func urlInstanceTypeDetect(r *http.Request) (instancetype.Type, error) {
 	return instancetype.Any, nil
 }
 
+// Pull out the Instance in order to filter
+func evaluateFieldInstanceFull(entry FilterEntry, instFull *api.InstanceFull) bool {
+	return evaluateFieldInstance(entry, &instFull.Instance)
+}
+
+
+// Evaluate a field of a filter on an instance
+func evaluateFieldInstance(entry FilterEntry, container *api.Instance) bool {
+	result := false
+	field := entry.Field
+	value := entry.Value
+	switch {
+		case strings.EqualFold(field, "name"):
+			logger.Warnf("In name eval, %s == %s", value, container.Name)
+			result = value == container.Name
+			break
+
+		case strings.EqualFold(field, "location"):
+			result = value == container.Location
+			break
+
+		case strings.EqualFold(field,"status") || strings.EqualFold(field, "state"):
+			result = container.Status == value
+			break
+
+		case strings.EqualFold(field,"type"):
+			result = container.Type == value
+			break
+
+		case strings.HasPrefix(field, "config"):
+			fieldCut := field[7:len(field)]
+			config := container.ExpandedConfig
+			result = config[fieldCut] == value
+			break
+
+		case strings.HasPrefix(field, "device"):
+			fieldSplit := strings.Split(field,".")
+			valSplit := strings.Split(value,".")
+			devices := container.ExpandedDevices
+			dev := devices[valSplit[0]]
+			result = dev != nil
+			if len(fieldSplit) > 1 && result {
+				result = dev[fieldSplit[1]] == valSplit[1]
+			}
+			break
```

Comment:
You should be able to use reflection to track down the struct field that matches the name provided and extract its type too. That would avoid having to effectively duplicate the entire struct definition in the filtering function.

Reply:
How would you suggest dealing with nested objects that require additional logic? We were thinking of hard coding a check to see if the object is a known nested object and then repeat the reflection. Does this make sense or is there a better way to do it?

Reply:
Yeah, with the data we're going to be feeding to this function, there isn't really much of a risk in doing recursion.

So long as we're dealing with structs, we can keep going deeper and deeper. If we hit a map, then we know we need to fetch the requested key from it. If we hit something else, then that's the end and we can compare.

It feels like a function that looks like `getFieldValue(input interface{}, key string) (value interface{}, error)` should be quite doable and could easily enough call itself to handle recursion, shortening the value of `key` by one level every time.

Reply:
Does that make sense?

Reply:
Makes sense, thank you!